// @vonage/vetch/dist/index.js
"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    function adopt(value) { return value instanceof P ? value : new P(function (resolve) { resolve(value); }); }
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : adopt(result.value).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.request = exports.instance = exports.Vetch = exports.ResponseTypes = exports.VetchPromise = exports.VetchError = void 0;
const vetch_1 = require("./vetch");
Object.defineProperty(exports, "Vetch", { enumerable: true, get: function () { return vetch_1.Vetch; } });
var types_1 = require("./types");
Object.defineProperty(exports, "VetchError", { enumerable: true, get: function () { return types_1.VetchError; } });
Object.defineProperty(exports, "VetchPromise", { enumerable: true, get: function () { return types_1.VetchPromise; } });
Object.defineProperty(exports, "ResponseTypes", { enumerable: true, get: function () { return types_1.ResponseTypes; } });
exports.instance = new vetch_1.Vetch();
function request(opts) {
    return __awaiter(this, void 0, void 0, function* () {
        return exports.instance.request(opts);
    });
}
exports.request = request;
